stdlib/public/runtime/ImageInspection.h

```diff
@@ -106,6 +109,37 @@ void addImageAccessibleFunctionsBlockCallbackUnsafe(const void *baseAddress,
 
 int lookupSymbol(const void *address, SymbolInfo *info);
 
+#if defined(_WIN32)
+/// Configure the environment to allow calling into the Debug Help library.
+///
+/// \param body A function to invoke. This function attempts to first initialize
+///   the Debug Help library. The result of that operation is passed to this
+///   function.
+///
+/// \returns Whether or not the Debug Help library was successfully initialized
+///   and \a body was subsequently invoked.
+///
+/// On Windows, the Debug Help library (DbgHelp.lib) is not thread-safe. All
+/// calls into it from the Swift runtime and stdlib should route through this
+/// function.
+SWIFT_RUNTIME_STDLIB_SPI
+bool _swift_withDebugHelpLibrary(
```

Comment:
This function should be exported so that it can be used by other Swift libraries in the repo if needed. It's not sufficient for those libraries to do their own locking; they would need to use the same lock to ensure thread safety.